wrapper.go

```diff
@@ -28,6 +29,48 @@ const (
 	HSMAutoDeprecated = "hsm-auto"
 )
 
+// KeyType defines types of cryptographic keys.
+type KeyType string
+
+const (
+	RSA2048 KeyType = "rsa_2048"
+	RSA3072 KeyType = "rsa_3072"
+	RSA4096 KeyType = "rsa_4096"
+)
+
+// Purpose defines the cryptographic capabilities of a key.
+type Purpose string
+
+const (
+	Encrypt Purpose = "encrypt"
+	Decrypt Purpose = "decrypt"
+	Sign    Purpose = "sign"
+	Verify  Purpose = "verify"
+	Wrap    Purpose = "wrap"
+	Unwrap  Purpose = "unwrap"
+)
+
+// ProtectionLevel defines where cryptographic operations are performed with a key.
+type ProtectionLevel string
+
+const (
+	Software ProtectionLevel = "software"
+	HSM      ProtectionLevel = "hsm"
+)
+
+// KMSKey represents a cryptographic key that can be imported into a KMS.
+type KMSKey struct {
+	Type            KeyType
+	Purposes        []Purpose
+	ProtectionLevel ProtectionLevel
+	Material        KeyMaterial
+}
+
+// KeyMaterial contains key material for various key types.
+type KeyMaterial struct {
+	RSAKey *rsa.PrivateKey
```

Comment:
Is this the only bit of information needed for key lifecycle operations that require key material?

Reply:
That's correct. Importing and rotating only require an `*rsa.PrivateKey` now. When more key types become supported, they'll be added to this struct.